### PR TITLE
Improvements in 1.x branch

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- type-safe comparisons
- remove unnecessary parentheses
- short syntax in applied operation
- remove one time variables
- use doublequotes
- remove unnecessary else statements
- a typo

### Testing Instructions
Code review

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
side note: frankmayer do you have glip account?

**Comment by andrepereiradasilva** [comment]
side note: @frankmayer do you have glip account?

**Comment by frankmayer** [comment]
@andrepereiradasilva no, I think  tried once, but something was wrong and I couldn't create an account or something like that...

**Comment by andrepereiradasilva** [comment]
still side note: ok, if you manage to get one tell me, that are some cms performance things i would like to talk to you.

**Comment by frankmayer** [comment]
@andrepereiradasilva I'll try again. Have to go now, Later.

File: Password/Simple.php [hunk 1]
```diff
@@ -126,22 +126,14 @@ protected function getSalt($length)
 	public function verify($password, $hash)
 	{
 		// Check if the hash is a blowfish hash.
-		if (substr($hash, 0, 4) == '$2a$' || substr($hash, 0, 4) == '$2y$')
+		$substrHash = substr($hash, 0, 4);
+		if ($substrHash === '$2a$' || $substrHash === '$2y$')
```
Comment (andrepereiradasilva) [inline Password/Simple.php]: cs: add a space before the if

File: Cipher/Simple.php [hunk 2]
```diff
@@ -194,7 +194,7 @@ private function _hexToInt($s, $i)
 				$k += 0;
 				break;
 			default:
-				$k = $k + (16 * (int) $c);
+				$k += (16 * (int) $c);
```
Comment (andrepereiradasilva) [inline Cipher/Simple.php]: remove the additional parenthisis

Comment (frankmayer) [inline Cipher/Simple.php]: nice, thanks

**Comment by andrepereiradasilva** [comment]
ok on code review

**Comment by frankmayer** [comment]
@andrepereiradasilva I am on glip now. What next?

**Comment by andrepereiradasilva** [comment]
add me andre.pereiradasilva@community.joomla.org

**Pull request merged**